package dev.shiwa.onearunbackend.controller

import dev.shiwa.onearunbackend.model.session.*
import dev.shiwa.onearunbackend.service.SessionService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/api/v1")
class SessionController(
    private val sessionService: SessionService,
) {
    @GetMapping("/session")
    fun getSession(@RequestParam @Valid sessionId: String): ResponseEntity<RaceSession> {
        val response = sessionService.getSession(sessionId)
        return ResponseEntity(response, HttpStatus.CREATED)
    }
    @PostMapping("/session")
    fun createSession(@RequestBody @Valid request: CreateSessionRequest): ResponseEntity<CreateSessionResponse>  {
        val response = sessionService.create(request.distanceInMeters)
        return ResponseEntity(response, HttpStatus.CREATED)
    }
    @PostMapping("/join")
    fun join(@RequestBody @Valid request: JoinSessionRequest): ResponseEntity<JoinSessionResponse>{
        val response = sessionService.join(request.joinCode)
        return ResponseEntity(response, HttpStatus.OK)
    }
}